java/query/query-core/src/main/java/sleeper/query/core/recordretrieval/QueryExecutor.java

```diff
@@ -103,13 +116,27 @@ public void initIfNeeded(Instant now) throws StateStoreException {
         init(now);
     }
 
+    /**
+     * Initialises a query executor with the partitions and partition to file mapping and set the next initialise time
+     * to now.
```

Comment:
This is incorrect, it doesn't set the next initialise time to now.

We could note that the partitions and partition to file mapping are loaded from the state store.